src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFOULT_SIZE = 10;
+    private int size;
+    private Object[] values;
+
+    public ArrayList() {
+        values = new Object[DEFOULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (values.length == size) {
+            resize();
+        }
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index != size) {
+            checkIndex(index);
+        }
+        if (++size == values.length) {
+            resize();
+        }
```

Comment:
the same here. you  can simplify  this conditionals

Reply:
It`s unworking wiitout it condition